### DISCUSSION TITLE
move the external library management from user area into admin area

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#7048] [General]
#6912 let me ask: Maybe it would be better to move the library management from user area into admin area

Currently external libraries are manged in different places
- mount possible folders in the docker compose file
- enable possible usage of external libraries in the admin area
- define and manage external libraries per user in user settings

issues:
- defining and managing external libraries is very technical. And maybe is not very family friendly (How to explain to a non-nerd wife o to my children?)
- it is not possible (or is it?) to avoid user A to add assets of user B as external libraries (and vice versa)

Moving external library management into the admin area would solve both issues.

**Comment by bo0tzz** [comment]
Previously suggested in https://github.com/immich-app/immich/discussions/4500#discussioncomment-7892234

**Reply by aisbergde** [reply]
then this idea here can be closed

**Reply by mich2k** [reply]
An ever simpler approach would be that a list of paths is sent from the admin to the user (also comma separated)

instead of: `/external/` (fixed unique path) we would have `/external/userA,/external/familyphotos,/external/commonmovies` from the admin page

the user, eventually, will be only be able to add paths that are present in that list as its external

This would be literally ten lines of code, the issue is:
if two users add a common path (like `/external/commonmovies`), will the thumbnails and everything be computed twice? Or there would be some shared metadata (face recognitions, thumbs etc) ?

As of today, the only guarantee a user will not add other user's data to his immich page is the hope that he doesn't know the name of other users' data path

edit: to prevent further issues/PRs that comment should be a feature request
in my opinion should be also performed some front end input/regex validation about the paths
and, instead of comma separated, the same “Edit import paths” mechanism/UI could be recycled for this
